src/docker-in-docker/install.sh

```diff
@@ -469,32 +469,24 @@ if [ "${INSTALL_DOCKER_BUILDX}" = "true" ]; then
     find "${docker_home}" -type d -print0 | xargs -n 1 -0 chmod g+s
 fi
 
-if [ -z "$DISABLE_IP6_TABLES" ]; then
-    DOCKER_DEFAULT_IP6_TABLES=""
-else
-    version_to_check=""
-    semver_regex="^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(?:-([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?(?:\+([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?$"
-    if echo "$DOCKER_VERSION" | grep -Eq "$semver_regex"; then
-        version_to_check=$(echo $DOCKER_VERSION | cut -d. -f1)
-    elif echo "$DOCKER_VERSION" | grep -Eq "^-?[0-9]+$"; then
-        version_to_check=$DOCKER_VERSION
-    fi
+DOCKER_DEFAULT_IP6_TABLES=""
+version_to_check=""
+semver_regex="^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(?:-([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?(?:\+([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?$"
+if echo "$DOCKER_VERSION" | grep -Eq "$semver_regex"; then
```

Comment:
Can you add a comment on what's happening here?

Reply:
comment added.